share/spack/docker/handle-prompt.sh

```diff
@@ -61,11 +61,11 @@ __write() {
 }
 
 __revparse_head() {
-    head="`git -C /spack rev-parse $@ HEAD 2>/dev/null`"
+    head="`git -C "$SPACK_ROOT" rev-parse $@ HEAD 2>/dev/null`"
```

Comment:
Instead of fancy quote handlings I am a big fan of proper variable notations, if that helps: `${SPACK_ROOT}`
```suggestion
    head="$(git -C ${SPACK_ROOT} rev-parse $@ HEAD 2>/dev/null)"
```

Reply:
The original syntax was chosen for POSIX compliance.

Reply:
oh, didn't know that syntax breaks it. good to know

Reply:
I'm confused about why @ax3l's suggestion isn't POSIX compliant.  I agree we need quotes for this.  But the use of `$()` is POSIX-compliant and honestly preferred over un-nestable backticks.